### PR TITLE
Improve BridgeClient error reporting

### LINKED ISSUE #227: Improve BridgeClient checking and error reporting
The BridgeClient should validate the resources folder it passes to the Bridge exists.  It should also capture the content in any non-200 response from the Bridge and include that information in any exception it throws to fail the client tests.

### PR DESCRIPTION
**Pull request by roncain**
Adds valiation of some of the configuration
properties before they are sent to the Bridge
so failure is detected earlier.

Also captures any content body returned from
the Bridge for any non-200 response so it can
be part of the exception thrown.  This makes it
easier to see what the Bridge detected was wrong.

Fixes #227

### CODE REVIEW

**Comment by roncain** [comment]
This PR is not ready for public review yet -- I'm verifying behavior in NET Native

**Comment by StephenBonikowsky** [comment]
Looks like there is a merge conflict.
Other than that, :shipit:

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs [hunk 1]
```diff
@@ -21,20 +22,26 @@ private static string BridgeBaseAddress
             get { return TestProperties.GetProperty(TestProperties.BridgeUrl_PropertyName); }
         }
 
-        static BridgeClient()
+        private static void EnsureBridgeIsRunning()
         {
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: What was the reason for moving from a static constructor to a static method?

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs [hunk 2]
```diff
@@ -58,7 +65,18 @@ private static void MakeConfigRequest()
...
                     var response = httpClient.PostAsync("/config/", content).Result;
                     if (!response.IsSuccessStatusCode)
-                        throw new Exception("Unexpected status code: " + response.StatusCode);
+                    {
+                        string reason = String.Empty;
+                        if (response.Content != null)
+                        {
+                            string contentAsString = response.Content.ReadAsStringAsync().Result;
+                            reason = String.Format("{0}Bridge returned content:{0}{1}",
+                                                    Environment.NewLine, contentAsString);
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: You've dropped the status code. We should have the status code and the status reason included.

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs [hunk 3]
```diff
@@ -58,7 +65,18 @@ private static void MakeConfigRequest()
...
+                        }
+                        throw new Exception(
+                            String.Format("Bridge returned unexpected status code {0}{1}", 
+                                            response.StatusCode, reason));
+                    }
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: There's also a status reason which is a textual description which might be helpful to include in the description.

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs [hunk 4]
```diff
@@ -58,7 +65,18 @@ private static void MakeConfigRequest()
                 {
                     var response = httpClient.PostAsync("/config/", content).Result;
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: GetAwaiter().GetResult() to avoid AggregateException if not using await

**Comment by mconnew** [comment]
Pending a couple of minor changes, the overall change LGTM

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs (hunk 1 above)
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: Better behavior for error handling.  As a static ctor, we got TypeInitialization exceptions for the BridgeClient that sort of masked something as simple as the bridge not running.  With an explicit failure from EnsureBridgeIsRunning, the exception stack trace is more informative.

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs (hunk 2 above)
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: Good catch -- added.

**Comment by roncain** [comment]
I merged this PR into #238 because I needed these improvements to debug it.  So I am closing this PR because it is now redundant.  The comments made in this PR were applied to the BridgeClient in #238

**Pull request closed without merging**